scenarios/throughput_stress.go

```diff
@@ -41,10 +41,30 @@ const (
 )
 
 type tpsState struct {
-	// CompletedIteration is the iteration that has been completed.
-	CompletedIteration int `json:"completedIteration"`
-	// WorkflowCount is the total number of workflows that have been completed so far.
-	WorkflowCount int `json:"workflowCount"`
+	// IterationWatermark is the highest iteration that has been completed so far.
+	// This is used to resume the scenario at the iteration again.
+	IterationWatermark int `json:"iterationWatermark"`
+	// IterationWatermarkUpdatedAt is the time when the iteration watermark was last updated.
+	IterationWatermarkUpdatedAt *time.Time `json:"iterationWatermarkUpdatedAt"`
+	// IterationWatermarkWorkflowCount is the total number of completed workflows for the iteration watermark.
+	IterationWatermarkWorkflowCount int `json:"iterationWatermarkWorkflowCount"`
+
+	// All the other fields below are transient state. They are reset when the scenario is resumed.
+	// They are useful to observe the progress of the scenario.
+
+	// IterationsStarted is the number of iterations that have been started so far.
+	IterationsStarted int `json:"iterationsStarted"`
+	// LastIterationStarted is the most recent iteration that was started.
+	LastIterationStarted int `json:"lastIterationStarted"`
+	// LastIterationStartedAt is the time when the last iteration was started.
+	LastIterationStartedAt *time.Time `json:"lastIterationStartedAt"`
+
+	// IterationsCompleted is the number of iterations that have been completed so far.
+	IterationsCompleted int `json:"iterationsCompleted"`
+	// LastIterationCompletedAt is the time when the last iteration was completed.
+	LastIterationCompletedAt *time.Time `json:"lastIterationCompletedAt"`
+	// AllIterationsCompletedAt is the time when the all iterations were completed.
+	AllIterationsCompletedAt *time.Time `json:"allIterationsCompletedAt"`
```

Comment:
Added a few of these for debugging purposes. This might change again as we learn more.